src/clincoded/messaging/__init__.py

```diff
@@ -849,6 +850,100 @@ def delivery_callback(err, msg):
         return_object['message'] = 'Message delivery failed'
         return return_object
 
+# Publish GDM - send GDM snapshot data (message) to Data Exchange when GDM is published
+@view_config(route_name='publish-gdm', request_method='POST')
+def publish_gdm(request):
+    return_object = {'status': 'Error',
+        'message': 'Unable to deliver full GDM'}
+
+    # Store JSON-encoded content of data
+    try:
+        resultJSON = request.json
+
+    except Exception as e:
+        sys.stderr.write('**** publish-gdm: Error sending data to Data Exchange - data not in expected format ****\n')
+        return_object['message'] = 'Retrieved data not in expected format'
+        return return_object
+
+    # Construct message
+    try:
+        message = json.dumps(resultJSON, separators=(',', ':'))
+
+    except Exception as e:
+        sys.stderr.write('**** publish-gdm: Error sending data to Data Exchange - failed to build complete message ****\n')
+        if e.args:
+            return_object['message'] = e.args
+        else:
+            return_object['message'] = 'Failed to build complete message'
+        return return_object
+
+    return_object = {'status': 'Error',
+                    'message': message,
+                    'error': 'Unable to deliver full GDM'}
+
+    # Set snapshot uuid and publish date as message key
+    key = resultJSON['uuid'] + '-' + resultJSON['resource']['publishDate']
+
+    # Configure message delivery parameters
+    kafka_cert_pw = ''
+
+    if 'KAFKA_CERT_PW' in os.environ:
+        kafka_cert_pw = os.environ['KAFKA_CERT_PW']
+
+    kafka_conf = {'bootstrap.servers': 'localhost:9093',
+            'log_level': 0,
+            'compression.type': 'gzip',
+            'security.protocol': 'ssl',
+            'ssl.key.location': 'etc/certs/client.key',
+            'ssl.key.password': kafka_cert_pw,
+            'ssl.certificate.location': 'etc/certs/client.crt',
+            'ssl.ca.location': 'etc/certs/server.crt'},
```

Comment:
Since every time when we test locally, we need to change `kafka_conf` to include only `bootstrap.servers`, `log_level` and `compression.type` to the following:
```
 kafka_conf = {'bootstrap.servers': 'localhost:9093',
                             'log_level': 0,
                             'compression.type': 'gzip'}
```
How about we change kafka config here to only include these three keys, so that testing locally will not require code change? 

Because looks like later on we already have if block  ` if request.host != 'localhost:6543':` to handle production config and setup all configs in `kafka_config`. Or maybe I'm missing some cases in local where we'll need credentials like `security.protocol`, `ssl.key.password`, ...?

Reply:
I left the codes there because they are for testing/validating the client credentials locally.  In this case, we do not need to do that so just comment out for easy testing.  I have updated my doc to only comment out the lines if do not need to test the certificate.  Thanks.

Reply:
I see, so there's need to test authentications for Kafka locally as well. Sounds good, I'll go ahead approve this PR.